test/unit/uiautomator2-specs.js

```diff
@@ -13,6 +13,182 @@ chai.use(chaiAsPromised);
 
 describe('UiAutomator2', function () {
   const adb = new ADB();
+
+  describe('shouldUninstallServerPackages', function () {
+    let uiautomator2;
+    const serverApk = {
+      'appPath': 'path/to/appium-uiautomator2-server.apk',
+      'appId': 'io.appium.uiautomator2.server'
+    };
+    const serverTestApk = {
+      'appPath': 'path/to/appium-uiautomator2-server-test.apk',
+      'appId': 'io.appium.uiautomator2.server.test'
+    };
+    beforeEach(function () {
+      uiautomator2 = new UiAutomator2Server(log, {
+        adb,
+        tmpDir: 'tmp',
+        systemPort: 4724,
+        host: 'localhost',
+        devicePort: 6790,
+        disableWindowAnimation: false,
+        disableSuppressAccessibilityService: false
+      });
+    });
+    it('with newer servers are installed', function () {
+      uiautomator2.shouldUninstallServerPackages([
+        {
+          'installState': adb.APP_INSTALL_STATE.NEWER_VERSION_INSTALLED,
+          ...serverApk
+        },
+        {
+          'installState': adb.APP_INSTALL_STATE.NEWER_VERSION_INSTALLED,
+          ...serverTestApk
+        }
+      ]).should.eql(true);
+    }),
+    it('with newer server is installed but the other could be old one', function () {
+      // Then, enforce to uninstall all apks
+      uiautomator2.shouldUninstallServerPackages([
+        {
+          'installState': adb.APP_INSTALL_STATE.NEWER_VERSION_INSTALLED,
+          ...serverApk
+        },
+        {
+          'installState': adb.APP_INSTALL_STATE.OLDER_VERSION_INSTALLED,
+          ...serverTestApk
+        }
+      ]).should.eql(true);
+    }),
+    it('with newer server is installed', function () {
+      uiautomator2.shouldUninstallServerPackages([
+        {
+          'installState': adb.APP_INSTALL_STATE.SAME_VERSION_INSTALLED,
+          ...serverApk
+        },
+        {
+          'installState': adb.APP_INSTALL_STATE.SAME_VERSION_INSTALLED,
+          ...serverTestApk
+        }
+      ]).should.eql(false);
+    }),
+    it('with older servers are installed', function () {
+      // then, installing newer serves are sufficient.
+      uiautomator2.shouldUninstallServerPackages([
+        {
+          'installState': adb.APP_INSTALL_STATE.OLDER_VERSION_INSTALLED,
+          ...serverApk
+        },
+        {
+          'installState': adb.APP_INSTALL_STATE.OLDER_VERSION_INSTALLED,
+          ...serverTestApk
+        }
+      ]).should.eql(false);
+    }),
+    it('with no server are installed', function () {
+      uiautomator2.shouldUninstallServerPackages([
+        {
+          'installState': adb.APP_INSTALL_STATE.NOT_INSTALLED,
+          'appPath': 'path/to/appium-uiautomator2-server.apk',
```

Comment:
I assume these items should also be replaced by constants like it was done above